packages/components/src/box-control/stories/index.story.tsx

```diff
@@ -0,0 +1,76 @@
+/**
+ * External dependencies
+ */
+import type { Meta, StoryFn } from '@storybook/react';
+
+/**
+ * WordPress dependencies
+ */
+import { useState } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import BoxControl from '../';
+import type { BoxControlValue } from '../types';
+
+const meta: Meta< typeof BoxControl > = {
+	title: 'Components (Experimental)/BoxControl',
+	component: BoxControl,
+	argTypes: {
+		onChange: { action: 'onChange' },
+	},
+	parameters: {
+		controls: { expanded: true },
+		docs: { canvas: { sourceState: 'shown' } },
+	},
+};
+export default meta;
+
+const defaultSideValues = {
+	top: '10px',
+	right: '10px',
+	bottom: '10px',
+	left: '10px',
+};
+
+const Template: StoryFn< typeof BoxControl > = ( props ) => {
+	const [ values, setValues ] =
+		useState< BoxControlValue >( defaultSideValues );
```

Comment:
No need to use `BoxControlValue`, we can use directly

```suggestion
		useState< ( typeof props )[ 'values' ] >( defaultSideValues );
```

Reply:
Interesting! I tried `(typeof BoxControl)[ 'values' ]` without success. Not sure I understand why that didn't work as well 🤔

Reply:
`typeof BoxControl` would return the component type (I believe it would tell you if the component is a class component, a function component, a JSX intrinsic string...), but that type doesn't have the prop types as properties.

On the other hand, `typeof props` returns the computed type for the properties of the `BoxControl` component — and that type has the `value` property.

Reply:
Expanding on this, an alternative version is also to grab the type of the props by using the `React.ComponentProps` utility type, which requires as argument the component _type_ :

`React.ComponentProps< typeof BoxControl >[ 'values' ]`


I'm ok with keeping `( typeof props )[ 'values' ]`, but I hope this helps with understanding these types!